core/core.go

```diff
@@ -474,6 +474,23 @@ func (n *IpfsNode) loadBootstrapPeers() ([]peer.PeerInfo, error) {
 func (n *IpfsNode) loadFilesRoot() error {
 	dsk := ds.NewKey("/local/filesroot")
 	pf := func(ctx context.Context, k key.Key) error {
+		ds := n.Repo.Datastore()
+		if old, err := ds.Get(dsk); err == nil && old != nil {
+			_ = n.Pinning.Unpin(ctx, key.Key(old.([]byte)), true)
```

Comment:
if this is an error and it is silenced then we may have a pin-leak (and pins growing and growing).